File: java/src/com/esielkar/abstract_factory/AbstractFactoryExample.java
package com.esielkar.abstract_factory;

public class AbstractFactoryExample {

    public static void main(String[] args) {
        AbstractFactory factory1 = new ConcreteFactory1();
        AbstractFactory factory2 = new ConcreteFactory2();

        final ProductA productA1 = factory1.createProductA();
        final ProductA productA2 = factory2.createProductA();

        final ProductB productB1 = factory1.createProductB();
        final ProductB productB2 = factory2.createProductB();

        System.out.println("Factory: " + factory1);
        System.out.println(productA1);
        System.out.println(productB1);

        System.out.println("Factory: " + factory2);
        System.out.println(productA2);
        System.out.println(productB2);

    }

}

File: dart/design_patterns/bin/abstract_factory/abstract_factory_example.dart
import 'factories.dart';
import 'products.dart';

void main(List<String> args) {
  final AbstractFactory factory1 = ConcreteFactory1();
  final AbstractFactory factory2 = ConcreteFactory1();

  final ProductA productA1 = factory1.createProductA();
  final ProductA productA2 = factory2.createProductA();

  final ProductB productB1 = factory1.createProductB();
  final ProductB productB2 = factory2.createProductB();

  print("Factory: ${factory1.runtimeType}");
  print(productA1);
  print(productB1);
  print("Factory: ${factory2.runtimeType}");
  print(productA2);
  print(productB2);
}
